modules/prebidServerBidAdapter.js

```diff
@@ -42,6 +66,21 @@ config.setDefaults({
  * @property {string} [cookieSetUrl] url for cookie set library, if passed then cookieSet is enabled
  */
 function setS2sConfig(options) {
+  if (options.defaultVendor) {
+    let vendor = options.defaultVendor
+
+    if (availVendorDefaults.hasOwnProperty(vendor)) {
+      Object.keys(availVendorDefaults[vendor]).forEach(function(vendorKey) {
+        if (s2sDefaultConfig[vendorKey] === options[vendorKey]) {
```

Comment:
I read somewhere that you shouldn't write code as clever as you are, because debugging code is 2x more difficult than writing it. So if you write clever code, you by definition cannot debug it... :)

Ok it's probably an exaggeration but the point remains. It took me awhile to figure out why this works - it's because its compare `undefined` to `undefined` and so it's actually proceeding to copy to undefined key over. It's not necessarily wrong, it's just hard to read. At minimum it needs a comment, or ideally a nicer way to understand what's going on in code.

Reply:
Thanks for the feedback @mkendall07, I've just finished making a slight change to hopefully infer the logic better.  I also put in a comment as a backup.  Please let me know your thoughts.